pkg/controller/egress/store/egressgroup_test.go

```diff
@@ -0,0 +1,183 @@
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	types2 "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	"antrea.io/antrea/pkg/apiserver/storage"
+	"antrea.io/antrea/pkg/controller/types"
+)
+
+func TestWatchEgressGroupEvent(t *testing.T) {
+	egTypeInvalid := types.EgressGroup{}
+
+	egressName := "egress"
+	egressUID := types2.UID("uid")
+	eg1 := &types.EgressGroup{
+		SpanMeta:          types.SpanMeta{NodeNames: sets.NewString("node-local")},
+		UID:               egressUID,
+		Name:              egressName,
+		GroupMemberByNode: map[string]controlplane.GroupMemberSet{},
+	}
+
+	eg2 := &types.EgressGroup{
+		SpanMeta:          types.SpanMeta{NodeNames: sets.NewString("node1", "node2")},
+		UID:               egressUID,
+		Name:              egressName,
+		GroupMemberByNode: map[string]controlplane.GroupMemberSet{},
+	}
+
+	pod1 := &controlplane.PodReference{Name: "pod1", Namespace: "namespace1"}
+	eg3 := &types.EgressGroup{
+		SpanMeta: types.SpanMeta{NodeNames: sets.NewString("node-local")},
+		UID:      egressUID,
+		Name:     egressName,
+		GroupMemberByNode: map[string]controlplane.GroupMemberSet{
+			"node-local": controlplane.NewGroupMemberSet(&controlplane.GroupMember{Pod: pod1}),
+		},
+	}
+
+	tests := []struct {
+		name           string
+		fieldSelector  fields.Selector
+		operations     func(p storage.Interface)
+		expectedEvents []watch.Event
+	}{
+		{
+			name:          "Add event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(egTypeInvalid)
+				store.Create(eg1)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+			},
+		},
+		{
+			name:          "Delete event",
+			fieldSelector: fields.SelectorFromSet(fields.Set{"nodeName": "node-local"}),
+			operations: func(store storage.Interface) {
+				store.Create(eg1)
+				store.Update(eg2)
+			},
+			expectedEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: &controlplane.EgressGroup{}},
+				{Type: watch.Added, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+				{Type: watch.Deleted, Object: &controlplane.EgressGroup{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: egressName,
+						UID:  egressUID,
+					},
+					GroupMembers: nil,
+				}},
+			},
+		},
+		{
+			name:          "Modify event",
```

Comment:
better to test an event with DeletedGroupMembers too

Reply:
Updated.